docker-compose.yml

```
@@ -112,7 +112,7 @@ services:
       start_period: 60s
 
   core:
-    image: ort-server-core
+    image: ${ORT_SERVER_IMAGE_PREFIX-ghcr.io/eclipse-apoapsis/}ort-server-core:${ORT_SERVER_IMAGE_TAG:-main}
```

Didn't try to run it, but should this and others include the ":"?

```suggestion
    image: ${ORT_SERVER_IMAGE_PREFIX:-ghcr.io/eclipse-apoapsis/}ort-server-core:${ORT_SERVER_IMAGE_TAG:-main}
```

---

Actually, this was on purpose: `:-` replaces the variable if it is not set or null (or empty). `-` only replaces it if it is not set. With `:-` it would not be possible to set the prefix to an empty value when using locally built images, as it would then always fall back to the default.

---

I found the solution in [this Stackoverflow answer](https://unix.stackexchange.com/a/122848/89805) which has a nice table showing the different substitution options at the end. And while it might work, I have to add that I don't like that syntax, as obviously not many people are aware of it and the `-` can also appear in the default value which can be confusing.